src/components/views/elements/RoomAliasField.tsx

```diff
@@ -50,25 +53,38 @@ export default class RoomAliasField extends React.PureComponent<IProps, IState>
     }
 
     private asFullAlias(localpart: string): string {
-        return `#${localpart}:${this.props.domain}`;
+        const hashAlias = `#${ localpart }`;
+        if (this.props.domain) {
+            return `${hashAlias}:${this.props.domain}`;
+        }
+        return hashAlias;
+    }
+
+    private get domainProps() {
+        const { domain } = this.props;
+        const prefix = <span>#</span>;
+        const postfix = domain ? (<span title={`:${domain}`}>{ `:${domain}` }</span>) : <span />;
+        const maxlength = domain ? 255 - domain.length - 2 : 255 - 1;   // 2 for # and :
+        const value = domain ?
+            this.props.value.substring(1, this.props.value.length - this.props.domain.length - 1) :
+            this.props.value.substring(1);
+
+        return { prefix, postfix, value, maxlength };
     }
 
     render() {
-        const poundSign = (<span>#</span>);
-        const aliasPostfix = ":" + this.props.domain;
-        const domain = (<span title={aliasPostfix}>{ aliasPostfix }</span>);
-        const maxlength = 255 - this.props.domain.length - 2;   // 2 for # and :
+        const { prefix, postfix, value, maxlength } = this.domainProps;
```

Comment:
Failing to follow why this got broken out into `domainProps`, especially things like the prefix which are just a const regardless. Is this class component going to get extended by another in the future?

Reply:
@t3chguy I moved it around as in my opinion this makes it more readable - you have a much cleaner reference where do does things come from and how they are referenced in a render without cluttering the render function much more.